.github/workflows/test-generator.yml

```diff
@@ -0,0 +1,60 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Test Library Generation
+
+# Uses protos from HEAD of https://github.com/googleapis/google-cloudevents
+# and tooling from HEAD of this repository to update all generated code.
+#
+# It creates a PR whose branch will include an unmodified snapshot of the
+# tooling used to generate changes.
+
+on:
+  pull_request:
+    paths:
+      - 'tools/**'
+      - 'generate-code.sh'
+      - '.github/workflows/test-generator.yml'
+
+jobs:
+  # Create a pull request.
+  test-generator:
+    runs-on: ubuntu-latest
+    env:
+      GENERATE_DATA_SOURCE: protos/google-cloudevents
+
+    steps:
+    - name: Go Library > Checkout Repository
+      uses: actions/checkout@v3
+    
+    - name: Proto Schemas > Checkout Repository
+      uses: actions/checkout@v3
+      with:
+        repository: googleapis/google-cloudevents
+        path: ${{ env.GENERATE_DATA_SOURCE }}
+
+    - name: Setup Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.19.x
+
+    - name: Install the generator
+      run: |
+        bash tools/setup-generator.sh
+
+    - name: Run the generator
+      run: |
+        bash ./generate-code.sh
```

Comment:
Is this just testing that it will run?

Reply:
Yes, if it completes without error we are assume the change didn't break the generator. It would be nice to do something like compare count of protos in to count of data.pb.go files out but I figured that could be added later... like permissions.